### PR TITLE
syncer: drain the old tick before Reset

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

since Reset will not clear the `ticker.C`, we manually do this

### Check List 

Tests 

 - pass original tests

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Review by glorv — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by Ehco1996 — APPROVED** [review]

File: syncer/syncer.go [hunk 1]
```diff
@@ -1341,14 +1341,20 @@ func (s *Syncer) syncDML(
 	defer ticker.Stop()
 	for {
 		// resets the time interval for each loop to prevent a certain amount of time being spent on the previous ticker
-		// execution to `executeSQLs` resulting in the next tikcer not waiting for the full waitTime.
+		// execution to `executeSQLs` resulting in the next ticker not waiting for the full waitTime.
```
Comment (gozssky) [inline syncer/syncer.go:1344]: It seems ticker is reset every time, why not use `time.Timer`?

Comment (lance6716) [reply syncer/syncer.go:1344]: https://github.com/pingcap/dm/pull/1939/commits/4bdb3cc848c92eae5f58009dd204aabfa4de5c99

**Review by gozssky — APPROVED** [review]

**Comment by Ehco1996** [comment]
> /merge

maybe we should change the PR title ?

**Review by gozssky — APPROVED** [review]

**Pull request merged**